src/index.js

```diff
@@ -0,0 +1,140 @@
+    } else if (tempNumber >= 50) {
+        className = 'greenTemp';
+    } else if (tempNumber <= 49) {
+        className = 'tealTemp';
+    }
+    return className
+}
+
+const getLandscape = (temperature) => {
+    if (temperature >= 80) {
+        return "🌵__🐍_🦂_🌵🌵__🐍_🏜_🦂";
+    } else if (temperature >= 70) {
+        return "🌸🌿🌼__🌷🌻🌿_☘️🌱_🌻🌷";
+    } else if (temperature >= 60) {
+        return "🌾🌾_🍃_🪨__🛤_🌾🌾🌾_🍃";
+    } else {
+        return "🌲🌲⛄️🌲⛄️🍂🌲🍁🌲🌲⛄️🍂🌲";
+    }
+};
+
+const getSky = (skyType) => {
+    switch(skyType) {
+        case 'sunny':
+            return "☁️ ☁️ ☁️ ☀️ ☁️ ☁️";
+        case 'cloudy':
+            return "☁️☁️ ☁️ ☁️☁️ ☁️ 🌤 ☁️ ☁️☁️";
+        case 'rainy':
+            return "🌧🌈⛈🌧🌧💧⛈🌧🌦🌧💧🌧🌧";
+        case 'snowy':
+            return "🌨❄️🌨🌨❄️❄️🌨❄️🌨❄️❄️🌨🌨";
+        default:
+            return '';
+    }
+};
+
+const getRealTemp = async (locationName) => {
+    const locationResp = await axios.get('http://localhost:5000/location',{ params: {q: locationName } })
+
+    const coords = [locationResp.data[0]['lat'], locationResp.data[0]['lon']]
+
+    const weatherResponse = await getWeather(coords[0],coords[1])
+
+    return weatherResponse
+
+}
+
+const getWeather = async (lat,lon) => {
+    const response = await axios
+        .get('http://localhost:5000/weather',{
+            params: {
+                lat: lat,
+                lon: lon,
+                units: 'imperial',
+            },
+        })
+
+        state.tempNumber = response.data['main']['temp']
+};
+
+const registerEventHandlers = () => {
+    state.increaseTempButton.addEventListener('click', () => {
+        state.tempNumberContainer.innerText = ++state.tempNumber;
+        state.tempNumberContainer.className = getTempColor(state.tempNumberContainer.innerText);
+        state.landEmojiContainer.innerText = getLandscape(state.tempNumber);
+    });
+    state.decreaseTempButton.addEventListener('click', () => {
+        state.tempNumberContainer.innerText = --state.tempNumber;
+        state.tempNumberContainer.className = getTempColor(state.tempNumber);
+        state.landEmojiContainer.innerText = getLandscape(state.tempNumber);
+    });
+
+    state.cityInput.addEventListener('input', () => {
+        state.cityNameContainer.innerText = state.cityInput.value;
+        state.weatherCity.innerText = state.cityInput.value;
+    });
+
+    state.realTempButton.addEventListener('click', async () => {
+        await getRealTemp(state.cityNameContainer.innerText)
+        state.tempNumberContainer.innerText = state.tempNumber;
+        state.tempNumberContainer.className = getTempColor(state.tempNumber);
+        state.landEmojiContainer.innerText = getLandscape(state.tempNumber);
+    });
+    state.skyDropdown.addEventListener('change', () => {
+        state.skyEmojiContainer.innerText = getSky(state.skyDropdown.value);
+    });
+};
+
+const onLoaded = () => {
+    loadControls();
+    registerEventHandlers();
+};
+
+onLoaded();
```

Comment:
I would suggest putting your `onLoaded` method inside an event handler for when the document is loaded, that way you can have an extra layer of protection when it comes to things loading in order. Basically, it would help ensure the `loadControls` and. `registerEventHandlers` functions would run after the DOM elements are loaded. It would look something like this:

```javascript
document.addEventListener("DOMContentLoaded", function () {
    loadControls()
    registerEventHandlers()
}

```

Reply:
Awesome job everyone! You really ate this one up! Please feel free to reach out to me if you have any questions about the comments I left or if you want to discuss anything in greater detail! ⭐️